### PR TITLE
Implement file soft deletion, allow deleting files upon reaching fail…

### PR DESCRIPTION
**Pull request by matszczygiel**
…ed, rejected or completed state.

### CODE REVIEW

File: drop-storage/migrations/003-retries/up.sql [hunk 1]
```diff
@@ -43,3 +43,7 @@ CREATE TABLE IF NOT EXISTS sync_incoming_files_inflight (
   FOREIGN KEY(sync_id, path_id) REFERENCES sync_incoming_files(sync_id, path_id) ON DELETE CASCADE ON UPDATE CASCADE
 );
 
+
+ -- paths soft deletion
+ALTER TABLE incoming_paths ADD COLUMN is_deleted INTEGER NOT NULL DEFAULT FALSE;
+ALTER TABLE outgoing_paths ADD COLUMN is_deleted INTEGER NOT NULL DEFAULT FALSE;
```
Comment (LukasPukenis) [inline drop-storage/migrations/003-retries/up.sql:49]:
```suggestion
ALTER TABLE incoming_paths ADD COLUMN is_deleted INTEGER NOT NULL DEFAULT 0 CHECK (is_deleted IN (0, 1));
ALTER TABLE outgoing_paths ADD COLUMN is_deleted INTEGER NOT NULL DEFAULT 0 CHECK (is_deleted IN (0, 1));
```

**Review by LukasPukenis — DISMISSED** [review]
Looks like a solid PR

File: drop-storage/migrations/003-retries/up.sql (hunk 1 above)
Comment (matszczygiel) [reply drop-storage/migrations/003-retries/up.sql:49]: Do you see any advantage of using `0 and 1` over `FALSE and TRUE`? It states intent better IMHO

Comment (LukasPukenis) [reply drop-storage/migrations/003-retries/up.sql:49]: I totally agree, it's just making sure that accidentally we wouldn't put in a non [0;1] number in there in some query. Maybe doing `is_deleted IN (TRUE, FALSE)` would work better but that's a speculation from my end :) You can freely dismiss this comment

Comment (matszczygiel) [reply drop-storage/migrations/003-retries/up.sql]: Done

**Pull request merged**